docs/tutorials/javascript/trade-on-ledger/use-amm-auction-slot-for-lower-fees.md

```diff
@@ -0,0 +1,221 @@
+
+## Appendix: AMM Formulas
+
+The above sample code shows how to use various AMM-related formulas in a particular workflow. Your code must also have implementations of those formulas, which can be in the same file or imported from a separate one as you desire.
+
+The file {% repo-link path="_code-samples/auction-slot/js/amm-formulas.js" %}`amm-formulas.js`{% /repo-link %} contains the source code for all these formulas.
+
+### AssetIn
+
+The _AssetIn_ formula calculates how much of an asset must be deposited to receive a target amount of LP Tokens.
+
+The following function implements AssetIn in JavaScript:
+
+{% code-snippet file="/_code-samples/auction-slot/js/amm-formulas.js" language="js" from="/* Implement the AMM single-asset deposit" before="/* Calculate how much to deposit" /%}
+
+This function depends on the `feeMult`, `feeMultHalf`, and `solveQuadraticEq` helper functions.
+
+### AuctionDeposit
+
+The _AuctionDeposit_ function is an inverted form of the formula for the AMM's auction price, taking into account how the minimum bid value changes as LP Tokens are issued. AuctionDeposit calculates the total amount of LP Tokens you need to receive in a deposit if you want to match the auction price:
+
+{% code-snippet file="/_code-samples/auction-slot/js/amm-formulas.js" language="js" from="/* Calculate how much to deposit" before="/* Calculate the necessary bid" /%}
+
+### AuctionPrice
+
+The _AuctionPrice_ function calculates the cost of the auction slot for current LP Token holders. **It is not used in this tutorial,** which assumes the user does not hold LP Tokens, but is presented here for completeness:
+
+{% code-snippet file="/_code-samples/auction-slot/js/amm-formulas.js" language="js" from="/* Calculate the necessary bid" before="module.exports =" /%}
+
+
+### SwapOut
+
+The _SwapOut_ formula, defined in the XRPL-0030 specification as formula 10, calculates how much of one asset you have to swap in to the AMM to receive a target amount of the other asset out from the AMM.
+
+The following function implements SwapOut in JavaScript:
+
+{% code-snippet file="/_code-samples/auction-slot/js/amm-formulas.js" language="js" from="/* Implement the AMM SwapOut formula" before="/* Compute the quadratic formula" /%}
+
+### feeDecimal, feeMult, and feeMultHalf
+
+These helper functions are used in other AMM formulas to convert from a trading fee value in the ledger (integer from 0 to 1000) to a decimal representation that can be multiplied by a total to apply the fee.
+
+The following functions implement `feeMult`, `feeMultHalf`, and `feeDecimal` in JavaScript:
+
+{% code-snippet file="/_code-samples/auction-slot/js/amm-formulas.js" language="js" from="/* Convert a trading fee to a value that" before="/* Implement the AMM SwapOut formula" /%}
+
+The AssetIn and SwapOut functions depend on these helper functions.
+
+### solveQuadraticEq
+
+This helper function is solves the quadratic equation, which you may remember from high-school mathematics.
```

Comment:
```suggestion
This helper function solves the quadratic equation, which you may remember from high-school mathematics.
```
Maybe remove the comment about high-school mathematics? How conversational are we trying to be?